audformat/core/utils.py

```diff
@@ -1265,6 +1267,60 @@ def union(
     if len(set(types)) != 1:
         objs = [to_segmented_index(obj) for obj in objs]
 
+    return union_misc(objs)
+
+
+def union_misc(
+    objs: typing.Sequence[pd.Index],
+) -> pd.Index:
+    r"""Create union of index objects.
+
+    Requires that levels and dtypes
+    of all objects match,
+    see :func:`audformat.utils.is_index_alike`.
+    Unlike :func:`audformat.utils.union`
+    index objects must not be conform to
+    :ref:`table specifications <data-tables:Tables>`.
+
+    Args:
+        objs: index objects
+
+    Returns:
+        union of index objects
+
+    Raises:
+        ValueError: if level and dtypes of objects do not match
+
+    Example:
+        >>> index1 = pd.Index([0, 1], name='idx')
+        >>> index2 = pd.Index([1, 2], dtype='Int64', name='idx')
+        >>> union_misc([index1, index2])
+        Index([0, 1, 2], dtype='Int64', name='idx')
+        >>> index3 = pd.MultiIndex.from_arrays(
+        ...     [['a', 'b', 'c'], [0, 1, 2]],
+        ...     names=['idx1', 'idx2'],
+        ... )
+        >>> index4 = pd.MultiIndex.from_arrays(
+        ...     [['b', 'c'], [1, 3]],
+        ...     names=['idx1', 'idx2'],
+        ... )
+        >>> union_misc([index3, index4])
+        MultiIndex([('a', 0),
+                    ('b', 1),
+                    ('c', 2),
+                    ('c', 3)],
+                   names=['idx1', 'idx2'])
+
+    """
+    if not objs:
+        return pd.Index([])
+
+    if not is_index_alike(objs):
```

Comment:
I would add a short cut before this line:

```suggestion
    if len(objs) == 1:
        return objs[0]
        
    if not is_index_alike(objs):
```

This requires that `objs` are only index objects, but you designed `audformat.utils.union_misc()` anyway to only support those. Was this on purpose? `audformat.utils.union()` also supports dataframes and series.

Reply:
done